site/content/extend/plugins/webapp/best-practices.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Design Best Practices
+date: 2019-06-02T00:00:00-00:00
+subsection: Web App Plugins
+weight: 0
+---
+
+
+### Actions that apply to specific Channels
+- Recommendation: Have your plugin register the actions to [the channel header](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerChannelHeaderButtonAction). This makes it quickly accessible by users and actions apply to the channel they're viewing. 
+- Example:  Zoom meeting posts to a channel
+
+![Custom Channel Header Button](/img/extend/bp-channel-header.png)
+
+You can additionally [register a slash command](https://developers.mattermost.com/extend/plugins/server/reference/#API.RegisterCommand) on the server-side to take channel-specific actions.
+- Example: Jira project subscriptions to a channel
+
+![Slash Command](/img/extend/bp-slash-command.gif)
+
+
+### Actions that apply to specific messages
+- Recommendation: Have your plugin register a [post dropdown menu component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerPostDropdownMenuComponent) with some text, icon and an action function. This adds your action to the "More Actions" post menu dropdown for easy discovery.
+- Examples: Create or attach to Jira issue from a message; copy a message to another channel; report an inappropriate message
+
+![Post Dropdown Menu](/img/extend/bp-post-dropdown-menu.png)
+
+### Actions related to files or images
+- Recommendation: Have your plugin register a [file upload method](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerFileUploadMethod) with some text, icon and an action function. This adds your new action to the file upload menu.
+- Examples: File sharing from OneDrive or GDrive; Draw plugin for sketches
+
+![File Upload Action](/img/extend/bp-file-upload.png)
+
+
+### Actions that apply to specific teams
+- Recommendation: Have your plugin register [left sidebar header component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerLeftSidebarHeaderComponent) with some text, icon and an action function. This adds your action above your team's channels in the sidebar.
+- Examples: Trello kanban board plugin, Github Plugin
```

Comment:
```suggestion
- Examples: Trello kanban board plugin, GitHub Plugin
```

Reply:
Actually, wondering if the GitHub plugin applies to specific teams? It might be an example of `Global actions that can be taken anywhere in the server and not directly related to teams, channels or users`